spec/datadog/core/remote/client/capabilities_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'datadog/core/remote/client/capabilities'
+require 'datadog/appsec/configuration'
+
+RSpec.describe Datadog::Core::Remote::Client::Capabilities do
+  subject(:capabilities) { described_class.new(settings) }
+
+  before do
+    capabilities
+  end
+
+  context 'when no component enabled' do
+    let(:settings) do
+      appsec_settings = Datadog::AppSec::Configuration::Settings.new
+      dsl = Datadog::AppSec::Configuration::DSL.new
+      dsl.enabled = false
+      appsec_settings.merge(dsl)
+
+      settings = Datadog::Core::Configuration::Settings.new
+      expect(settings).to receive(:appsec).at_least(:once).and_return(appsec_settings)
```

Comment:
@lloeki I believe there is some leak going on on our testing suite for `dd-trace-rb`

I added a note for me to take a look after we release the latest version, but for now, the quickest way to fix it was making sure this test state stays correct, rather than fixing the original leak